src/api/spec/models/user_spec.rb

```diff
@@ -804,6 +744,59 @@
     end
   end
 
+  shared_examples 'password comparison' do
+    context 'with invalid credentials' do
+      it 'returns false' do
+        expect(user.authenticate('invalid_password')).to eq(false)
+      end
+    end
+
+    context 'with valid credentials' do
+      it 'returns a user object for valid credentials' do
+        expect(user.authenticate('opensuse')).to eq(user)
+      end
+    end
+  end
+
+  describe '#authenticate' do
+    context 'as a user which has a deprecated password' do
+      let(:user) { create(:user_deprecated_password) }
+
+      context 'conversation of deprecated password' do
+        before do
+          user.authenticate('opensuse')
+        end
+
+        it 'converts the password to bcrypt' do
+          expect(BCrypt::Password.new(user.password_digest).is_password?('opensuse')).to be_truthy
+        end
+
+        it 'resets the hash of the deprecated password' do
+          expect(user.deprecated_password).to be(nil)
+        end
+
+        it 'resets the hash type of the deprecated password' do
+          expect(user.deprecated_password_hash_type).to be(nil)
+        end
+
+        it 'resets the salt of the deprecated password' do
+          expect(user.deprecated_password_salt).to be(nil)
+        end
+      end
+
+      it_behaves_like 'password comparison'
+    end
+
+    context 'as a user which has a bcrypt password' do
+      before do
+        user.password = 'opensuse'
```

Comment:
[nitpick] If you use buildservice as password in your introduced factory you don't need to change it here and you we're also consistent.